TestResources_v3/Custom Assertions/AssertJSONEqual.swift

```diff
@@ -18,20 +18,48 @@ func error(domain: String, code: Int = -1, message: @autoclosure () -> String) -
 ///   - expression2: JSON object 2, as Data.
 ///   - file: file the assert is being made
 ///   - line: line the assert is being made.
-func AssertJSONEqual(_ expression1: @autoclosure () throws -> Data,
-                     _ expression2: @autoclosure () throws -> Data,
-                     file: StaticString = #file,
-                     line: UInt = #line) {
+func AssertJSONEqual(
+  _ expression1: @autoclosure () throws -> Data,
+  _ expression2: @autoclosure () throws -> Data,
+  file: StaticString = #file,
+  line: UInt = #line
+) {
   do {
     guard let json1 = try JSONSerialization.jsonObject(with: expression1()) as? [String: Any] else {
       throw error(domain: "AssertJSONEqual", message: "First expression is not a valid json object!")
     }
     guard let json2 = try JSONSerialization.jsonObject(with: expression2()) as? [String: Any] else {
       throw error(domain: "AssertJSONEqual", message: "Second expression is not a valid json object!")
     }
+
+    AssertJSONEqual(json1, json2, file: file, line: line)
+
+  } catch {
+    XCTFail("Error: \(error)", file: file, line: line)
+  }
+}
+
+/// Asserts the given 2 JSON Serializations are equal, by creating JSON objects from Data and comparing dictionaries.
+/// Recursively calls itself for nested dictionaries.
+/// - Parameters:
+///   - expression1: JSON object 1
+///   - expression2: JSON object 2
+///   - file: file the assert is being made
+///   - line: line the assert is being made.
+func AssertJSONEqual(
+  _ expression1: @autoclosure () throws -> [String: Any],
+  _ expression2: @autoclosure () throws -> [String: Any],
```

Comment:
These shouldn't be `@autoclosure throws` IMHO, the thrown errors are already handled above

Reply:
But you can also call it directly.